include/modules.h

```diff
@@ -118,6 +120,62 @@ struct mapi_mheader_av2
 #define DECLARE_MODULE_AV2(name, reg, unreg, cl, hl, hfnlist, caplist, v, desc) \
 	struct mapi_mheader_av2 _mheader = { MAPI_V2, reg, unreg, cl, hl, hfnlist, caplist, v, desc, DATECODE}
 
+
+/***
+Version 3 modules utilize a flexible key/value vector.
+
+Example:
+DECLARE_MODULE_AV3
+(
+	MOD_ATTR { "name",    "mymodule"                       },
+	MOD_ATTR { "mtab",    &message_foo                     },
+	MOD_ATTR { "mtab",    &message_unfoo                   },
+	MOD_ATTR { "init",    modinitfunc                      },
+	MOD_ATTR { "hook",    MOD_HOOK   { "myhook", &id }     },
+	MOD_ATTR { "hookfn",  MOD_HOOKFN { "myhook", hookfun } },
+)
+
+Notes:
+- Multiple keys with the same name will have different behavior depending on the logic for that key.
+- On load, the order in which keys are specified is the order they will be evaluated (top to bottom).
+- On unload, the evaluation is the REVERSE order (bottom to top).
+- If an init function returns false, or other error occurs, no further keys are evaluated and the
+  unload rolls back from that point.
+***/
+
+struct mapi_av3_attr
+{
+	#define MAPI_V3_KEY_MAXSZ 16
```

Comment:
This is just for the length of the string of the key, for strncmp(), so I don't have to use strcmp() - habit for not setting off static code analyzers, and maybe even useful to not crash the IRCd from a corrupt module who knows. There's no limit on the number of keys. I'll rename it MAXLEN.